Mathlib/Algebra/CharP/Defs.lean

```diff
@@ -38,7 +38,7 @@ For instance, endowing `{0, 1}` with addition given by `max` (i.e. `1` is absorb
 This example is formalized in `Counterexamples/CharPZeroNeCharZero.lean`.
 -/
 @[mk_iff]
-class _root_.CharP (R : Type*) [AddMonoidWithOne R] (p : ℕ) : Prop where
+class _root_.CharP (R : Type*) [AddMonoidWithOne R] (p : semiOutParam ℕ) : Prop where
```

Comment:
Should the instance
```
instance charP : CharP R (ringChar R) :=
```
on line 147 of this file now become a lemma?
(I would be in favour.)

Reply:
Many things fail if we do this.